### ISSUE TITLE
Feedback from testing: Small updates to ADVS Template and Vignette

### ISSUE THREAD

**Issue by manciniedoardo** [pharmaverse/admiralmetabolic#67]
## Background Information

The updates below were suggested from package testing.

### Template
- [x] Include in the label line a mention of "for Metabolic Trials" or something to that effect so that it clearly distinguishes from the admiral core ADVS template
- [x] Add `ASEQ` derivation
- [x] Add derivations for `BASECAT1` and `BASECA1N` to match vignette.
- [x] Ensure there is only ever at most one empty line at a time, in line with [tidyverse style guide](https://style.tidyverse.org/syntax.html#vertical-space).

### Vignette
- [x] Remove `advs <- advs %>% filter(!VSTESTCD == "BMI")` because it violates the data minimisation guidance (don't collect data which is not required for the analysis). Furthermore, the code is error-prone. If another parameter was derived before BMI, it would be removed as VSTESTCD is NA for this parameter.
- [x] Remove first paragraph of "Derive Categorization Variables (AVALCATx, BASECATx)" (`admiral does not currently have a generic function to aid in assigning AVALCATy/ AVALCAvN and BASECATy/ BASECAvN values. Below is a simple example of how these values may be assigned:`) because it contradicts the second one (`For deriving categorization variables (AVALCATx, BASECATx) admiral provides derive_vars_cat()  (see documentation of the function for details).`)
- [x] Always use text `AVALCATy` and `BASECATy` instead of `AVALCATx` and `BASECATx` to be in line with CDISC terminology.

### Template and Vignette
- [ ] Replace `constant_by_vars = exprs(USUBJID)`  with `get_admiral_option("subject_keys")`
- [x] Consider using `derive_var_base()` (or `derive_vars_merged()`) for deriving `BASECATy` and `BASECAyN`. This avoids defining the categories twice.
- [x] Renaming: `CRITyFN` instead of `CRITyFLN`.

**Comment by AndersAskeland** [comment]
I don't agree with the "Not calculating BMI". In my view, BMI is not actually collected as such, but rather a derivation of weight and height (which is collected). Furthermore, as we are generating ADaM, I don't think the re-derivations are against any minimized collection conventions. 

However, I agree that the filter is error prone, which can be corrected with adding a `is.na()`.

One could argue that it should be removed from the SDTM dataset, but I think that showing how to remove it makes sense, as you might not necessarily have access to the SDTM when working on ADaM's
